lib/utils.js

```diff
@@ -156,17 +156,21 @@
 	var obj = Type.obj, obj_is = obj.is, obj_has = obj.has, obj_map = obj.map;
 
 	var Val = {};
-	Val.is = function(v){ DEP('val.is'); // Valid values are a subset of JSON: null, binary, number (!Infinity), text, or a soul relation. Arrays need special algorithms to handle concurrency, so they are not supported directly. Use an extension that supports them if needed but research their problems first.
-		if(v === u){ return false }
-		if(v === null){ return true } // "deletes", nulling out keys.
-		if(v === Infinity){ return false } // we want this to be, but JSON does not support it, sad face.
-		if(text_is(v) // by "text" we mean strings.
-		|| bi_is(v) // by "binary" we mean boolean.
-		|| num_is(v)){ // by "number" we mean integers or decimals.
-			return true; // simple values are valid.
-		}
-		return Val.link.is(v) || false; // is the value a soul relation? Then it is valid and return it. If not, everything else remaining is an invalid data type. Custom extensions can be built on top of these primitives to support other types.
-	}
+	// Valid values are a subset of JSON: null, binary, number (!Infinity), text,
+	// or a soul relation. Arrays need special algorithms to handle concurrency,
+	// so they are not supported directly. Use an extension that supports them if
+	// needed but research their problems first.
+	Val.is = (v) => {
+	  DEP("val.is");
+	  // "deletes", nulling out keys.
+	  v === null ||
+	  "string" === typeof v ||
+	  "boolean" === typeof v ||
+	  // we want +/- Infinity to be, but JSON does not support it, sad face.
+	  // can you guess what v === v checks for? ;)
+	  ("number" === typeof v && v != Infinity && v != -Infinity && v === v) ||
+	  (v && "string" == typeof v["#"] && Object.keys(v).length === 1 && v["#"]);
+	};
```

Comment:
i'm not even going to pretend like i know the value of this change. it seems big! haha

Reply:
Well, on line 171, we only check for `+/- Infinity` and `NaN` only if we deal with `number`
On line 172, we check if `v` is a soul (an object with a single property `#`), and if so, we return its value. This is old behavior